docs/api/1.x/synchronisation.rst

```diff
@@ -13,67 +13,133 @@ This section describes the basic aspects of synchronisation using *Kinto*.
+#. If response has ``Next-Page`` header, follow full URL in header.
+#. If list of changes is empty, **done** → up-to-date.
+
+.. image:: ../../images/sync-oldest.svg
+
+If an error occurs during the obtention of pages,
+the synchronisation can be resumed transparently, since the pages are obtained
+with ascending timestamps, and the next sync relies on the highest
+timestamp successfully stored locally.
+
+
+Strategy #2 — Newest first
+--------------------------
+
+In order to populate a UI, it might be relevant to obtain the latest changes first.
+
+Syncing newest records first is a bit more complex since changes can occur between
+the obtention of the first and the last pages.
+
+We will use ``sort=-last_modified`` (desc), ``_before`` to omit later changes,
+and ``_since`` to include changes after last sync:
+
+#. First sync: ``timestamp := 0``
+#. Next sync: use ``timestamp`` stored in last successful sync.
+#. Fetch current collection timestamp ``HEAD /buckets/<bid>/collections/<cid>/records``
+   in ``ETag`` response header and store its value in ``start``.
+#. Fetch ``GET /buckets/<bid>/collections/<cid>/records?_sort=-last_modified&_before=<start>&_since=<timestamp>``
+#. If response is ``200 OK``, stack the obtained list of remote changes.
+#. If response has ``Next-Page`` header, follow full URL in header.
+#. If list of changes is empty, **done** → handle the stack of remote changes
+   and update the timestamp: ``timestamp := MAX(local_records['last_modified'])``
+
+.. image:: ../../images/sync-newest.svg
+
+With this approach, the main algorithm is rather simple but since we track the
+*last sync timestamp* when the last page is done, if an error occurs
+between the first and the last step, the client must redownload every page obtained
+from *step 1* until it succeeds to fetch every page of the sync.
+
+In order to avoid that, the algorithm should slightly be complexified in order to
+track additional info obtained from the page that failed. The upper and lower
+values of timestamps (``_before`` and ``_since``) can then
+be specified manually to resume the synchronisation.
+
+
+Strategy #3 — Newest first, partially
+-------------------------------------
+
+For very large collections, it could be interesting to perform a first *partial*
+synchronisation, and then fetch old records in background.
+
+When a new client wants to sync, instead of syncing hundreds of pages on the
+first synchronization, two distinct synchronization processes can be combined.
+
+For example, start with some recent records in order to populate a UI,
+and then fetch older records in background.
+
+#. Obtain a few pages of recent records using the *newest first* strategy from above
+#. In background, fetch old records using ``_sort=-lastmodified`` and ``_before=MIN(local_records[last_modified])``
+#. Recent changes can be obtained using ``_sort=-lastmodified`` and ``_since=MAX(local_records[last_modified])``
```

Comment:
I still think you'll skip records with this strategy unless you can be sure that each record has a unique `lastmodified` timestamp. As soon as you have 1 local record with a given timestamp, all other remote records with that same timestamp will be ignored, right?